### ISSUE TITLE
insert_property_defaults does not work for nested schemas (eg. oneOf)

### ISSUE THREAD

**Issue by madejejej** [davishmcclurg/json_schemer#94]
Hi, first of all, thank you for this wonderful gem!

I tried to set up the smallest possible reproduction script. It seems that the workaround to this issue is to use a top-level `if-then-else` clause, but it makes it harder to use nested schemas.

```ruby
# frozen_string_literal: true

require 'json_schemer'
require 'json'

schema_def = {
  "$schema" => "http://json-schema.org/draft-07/schema#",
  "type" => "object",
  "oneOf" => [
    "$ref" => "#/definitions/a"
  ],
  "required" => ["field", "default_field"],
  "definitions" => {
    "a" => {
      "properties" => {
        "field" => { "type" => "string", "const" => "a" },
        "default_field" => { "enum" => ["f1", "f2"], "default" => "f1" },
      }
    }
  }
}.to_json

schema_def2 = {
  "$schema" => "http://json-schema.org/draft-07/schema#",
  "type" => "object",
  "required" => ["field", "default_field"],
  "properties" => {
    "field" => { "type" => "string", "const" => "a" },
    "default_field" => { "enum" => ["f1", "f2"], "default" => "f1" },
  }
}.to_json

schema = JSONSchemer.schema(schema_def, insert_property_defaults: true)
schema2 = JSONSchemer.schema(schema_def2, insert_property_defaults: true)

params = { "field" => "a" }
params2 = params.dup

schema.valid?(params) 
schema2.valid?(params2)
pp params # => {"field"=>"a"}
pp params2 # => {"field"=>"a", "default_field"=>"f1"}
```

**Comment by davishmcclurg** [comment]
Thanks for the report @madejejej! The behavior you're seeing is currently expected: https://github.com/davishmcclurg/json_schemer/pull/59

The tricky part here is knowing which default to use if multiple nested schemas are valid. For simplicity's sake, we decided to disable inserting defaults for nested schemas, but let me know if you can think of a better solution.

**Comment by davishmcclurg** [comment]
Released in 2.0.0. More info in the commit: https://github.com/davishmcclurg/json_schemer/commit/aaafab1b02e6017d3048ad62cfc125a33a5c217f
